src/main/java/King.java

```diff
@@ -0,0 +1,173 @@
+                tasks.get(parseTaskIndex(userInput)).markAsUndone();
+            } else if (userInput.startsWith("todo")) {
+                addToDoTask(userInput);
+            } else if (userInput.startsWith("deadline")) {
+                addDeadlineTask(userInput);
+            } else if (userInput.startsWith("event")) {
+                addEventTask(userInput);
+            } else {
+                addToDoTask(userInput);
+            }
+        }
+        exit();
+    }
+
+    public static void exit() {
+        System.out.println("____________________________________________________________\n");
+
+        if (repeatCount >= 5) {
+            System.out.println(" I am repeating no more!!!\n");
+        } else {
+            System.out.println(" Bye. Hope to see you again soon!\n");
+        }
+
+        System.out.println("____________________________________________________________");
+    }
+
+
+    public static int parseTaskIndex(String text) {
+        return Integer.parseInt(text.split(" ")[1]) - 1;
+    }
+
+
+    public static void printList() {
+        System.out.println("____________________________________________________________\n"
+                           + "Here are the tasks in your list:");
+
+        for (int i = 0; i < tasksCount; i++) {
+            int indexNum = i + 1;
+            System.out.println(indexNum + ". " + tasks.get(i));
+        }
+
+        System.out.println("____________________________________________________________");
+    }
+
+
+    public static void addToDoTask(String text) {
+        tasksCount += 1;
+        Task t = new Todo(text);
+        tasks.add(t);
+        printAddedTaskDescription(t);
+    }
+
+    public static void addDeadlineTask(String text) {
+        tasksCount += 1;
+        String[] taskSpecifics = text.split(" ");
+        int seperationIndex = 0;
+        String taskContent = "";
+        String taskEndTime = "";
+
+        for (int i = 0; i < taskSpecifics.length; i++) {
+            if (taskSpecifics[i].equals("/by")) {
+                seperationIndex = i;
+            }
+        }
+        for (int i = 1; i < seperationIndex; i++) {
+            taskContent += taskSpecifics[i];
+            if (i < seperationIndex - 1) {
+                taskContent += " ";
+            }
+        }
+        for (int i = seperationIndex + 1; i < taskSpecifics.length; i++) {
+            taskEndTime += taskSpecifics[i];
+            if (i < taskSpecifics.length - 1) {
+                taskEndTime += " ";
+            }
+        }
+
+        Task t = new Deadline(taskContent, taskEndTime);
+        tasks.add(t);
+        printAddedTaskDescription(t);
+    }
+
+    public static void addEventTask(String text) {
+        tasksCount += 1;
+        String[] taskSpecifics = text.split(" ");
+        int seperationIndexFirst = 0;
+        int seperationIndexSecond = 0;
+        String taskContent = "";
+        String taskStartTime = "";
+        String taskEndTime = "";
```

Comment:
Maybe use fewer variables, as it is hard to keep track of all these variables in the code flow.